### PR TITLE
Add shouldPostEvent check for JVMTI FramePop event 

### PR DESCRIPTION
**Pull request by thallium**
The shouldPostEvent check was missed for the JVMTI FramePop event in #16654.

### CODE REVIEW

**Comment by thallium** [comment]
@babsingh FYI

**Comment by babsingh** [comment]
LGTM.

nit: fix grammar, and make PR description and commit message consistent.

```
Add shouldPostEvent check for JVMTI FramePop event

The shouldPostEvent check was missed for the JVMTI FramePop event
in #16654.

Signed-off-by: Gengchen Tuo <gengchen.tuo@ibm.com>
```

**Comment by babsingh** [comment]
jenkins test sanity.functional,sanity.openjdk alinux64 jdk20

**Comment by babsingh** [comment]
jenkins compile win jdk17

**Comment by babsingh** [comment]
PR builds:
- https://openj9-jenkins.osuosl.org/job/Pipeline_Build_Test_JDK20_aarch64_linux/177/
- https://openj9-jenkins.osuosl.org/job/Pipeline_Build_Test_JDK17_x86-64_windows/1749/

File: runtime/jvmti/jvmtiHook.c [hunk 1]
```diff
@@ -2046,25 +2046,24 @@ jvmtiHookMonitorWaited(J9HookInterface** hook, UDATA eventNum, void* eventData,
 static void
 jvmtiHookFramePop(J9HookInterface** hook, UDATA eventNum, void* eventData, void* userData)
 {
-	J9JVMTIEnv * j9env = userData;
-	J9VMFramePopEvent * data = eventData;
+	J9JVMTIEnv *j9env = userData;
+	J9VMFramePopEvent *data = eventData;
 	jvmtiEventFramePop callback = j9env->callbacks.FramePop;
+	J9VMThread *currentThread= data->currentThread;
+	J9Method *method= data->method;
```
Comment (babsingh) [inline runtime/jvmti/jvmtiHook.c:2053]: missing space.
```suggestion
	J9VMThread *currentThread = data->currentThread;
	J9Method *method = data->method;
```

**Review by babsingh — APPROVED** [review]
Recent formatting changes do not require new PR builds. Approving based upon old PR builds: https://github.com/eclipse-openj9/openj9/pull/17604#issuecomment-1594035960.

**Pull request merged**